partitioner.go

```diff
@@ -129,6 +138,36 @@ func (p *Partition) HandleInRoundRobin(handler Handler) {
 	atomic.AddInt64(&p.messagesInFlight, 1)
 }
 
+// HandleDebounced debounce the handler high order function in round robin
+// handler: high order function to execute
+func (p *Partition) HandleDebounced(handler Handler, key string) {
+	// Backoff if currentInFlight > nPart * maxMessagesPerPartition
+	// this means messages are in error and or buffer is full
+	currentInFlight := atomic.LoadInt64(&p.messagesInFlight)
+	for currentInFlight >= int64(p.nPart)*int64(p.maxMessagesPerPartition) {
+		time.Sleep(2 * time.Millisecond)
+	}
+
+	p.Lock()
+	defer p.Unlock()
+	timer, found := p.debounceTimers[key]
+	if found {
+		timer.Stop()
```

Comment:
No because the inflight will be increased when we run AfterFunc, and decreased when the message is successfully processed. If I decrease here we have negative messages in flight since the timer never expired we never increased.